doc/api_base.rst

```diff
@@ -336,3 +336,5 @@
         base.download_packages(base.transaction.install_set, progress)
         print("Installing...")
         base.do_transaction()
+  .. method:: reboot_needed()
```

Comment:
We need to keep the whitespace formatting as with other methods. I tried to generate the docs, see here:

![reboot_needed_fix](https://user-images.githubusercontent.com/110612307/228241415-43285ad3-817b-4258-80f9-fc761292e069.png)

Reply:
Ah thanks, was missing that newline.

Converted this to a draft, I'm going to investigate using advisories to detect whether a reboot is recommended, and if that turns out to be a better way, then I'll revise this documentation.